src/nupic/regions/RecordSensor.py

```diff
@@ -548,8 +549,9 @@ def readFromProto(cls, proto):
     instance = cls()
 
     instance.encoder = MultiEncoder.read(proto.encoder)
-    instance.disabledEncoder = MultiEncoder.read(proto.disabledEncoder)
-    instance.topDownMode = proto.topDownMode
+    if proto.disabledEncoder is not None:
+      instance.disabledEncoder = MultiEncoder.read(proto.disabledEncoder)
+    instance.topDownMode = True if proto.topDownMode else False
```

Comment:
I believe bool(proto.topDownMode) will work here. And above you may be able to do int(self.topDownMode)